src/coupling/balance.cc

```diff
@@ -637,8 +590,13 @@ void Balance::calculate_cumulative(unsigned int quantity_idx,
 
 	// compute fluxes on boundary regions: R'.(F.u + f)
 	chkerr(VecZeroEntries(boundary_vec));
+    
 	Vec temp;
-	chkerr(VecDuplicate(ones_be_, &temp));
+    chkerr(VecCreateMPI(PETSC_COMM_WORLD,
+			be_regions_.size(),
+			PETSC_DECIDE,
+			&temp));
+    
 	chkerr(MatMultAdd(be_flux_matrix_[quantity_idx], solution, be_flux_vec_[quantity_idx], temp));
```

Comment:
to JB: this might be also implemented directly, without creating MPI objects and ```MatMultTranspose```.
```increment_fluxes_``` would be then computed locally and reduced in the output function similarly to ```increment_sources_```. Should be done now or be left for future?

Reply:
I did minor simplifications.
The larger changes, including boundary edge indexing, are described in #1070.